### DISCUSSION TITLE
How to style using CSS classes instead of props?

### DISCUSSION THREAD

**Discussion by tomaszferens** [chakra-ui/chakra-ui#3963] [Q&A]
I'm looking for an equivalent of [makeStyles](https://material-ui.com/styles/api/#makestyles-styles-options-hook) from @material-ui. 

So, instead of writing markup like this:

```jsx
function MyComponent() {
  return (
    <Box p={2} borderRadius={5} bg="white">
      {/* */}
    </Box>
  );
}
```

I want to do something like this:
```jsx
const useStyles = makeStyles({
  root: {
    padding: 2,
    borderRadius: 5,
    background: "white",
  },
});

function MyComponent() {
  const styles = useStyles();

  return <div className={styles.root}>{/* */}</div>;
}
```

Not sure how to achieve it in chakra.

**Comment by mihir-karbelkar-wiz** [comment]
Would be really helpful to reduce same styles being repeated again and again

Like if I have same box with same props, with my editor it'll move the code to three lines and if I have six such boxes that's 12 lines wasted.

**Comment by HaleyWardo** [comment]
You could use thier styled components.

```
import {chakra} from '@chakra-ui/react';

const Container = chakra('div', {
  baseStyle: {
    padding: 2,
    borderRadius: 5,
    background: "white",
  }
 }
)

function MyComponent() {
  return <Container>{/* */}</Container>;
}

```
